### PR TITLE
WIP: Run node 10 with '--experimental_worker' flags

### PR DESCRIPTION
**Pull request by jeanfabrice**
> Please provide enough information so that others can review your pull request:

Fixes: #4335 for both Docker and Linux setup:
- add `--experimental-worker` to Dockerfile
- move common constants and functions to `functions.sh` adding a new `compute_node_args` that may compute additionals node arguments based on several tests. Here, use this function to add `--experimental-worker` when node 10 is detected.
- source `functions.sh` in all startup scripts, and use `compute_node_args` as a node argument

I did not check Windows environment since I don't have any

### CODE REVIEW

File: bin/functions.sh [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+fatal() { error "$@"; exit 1; }
+is_cmd() { command -v "$@" >/dev/null 2>&1; }
+
+compute_node_args() {
+  NODE_VERSION_STRING=$(node --version)
+  NODE_VERSION_STRING=${NODE_VERSION_STRING#"v"}
+  DETECTED_MAJOR=$(pecho "$NODE_VERSION_STRING" | cut -s -d "." -f 1)
+  ARGS=""
+
+  [ -n "$DETECTED_MAJOR" ] || fatal "Cannot extract $PROGRAM_LABEL major version from version string \"$VERSION_STRING\""
```
Comment (rhansen) [inline bin/functions.sh:21]: This logic is repeated in `require_minimal_version`. Please move it to a separate function that is called by both `compute_node_args` and `require_minimal_version`.

**Review by rhansen — COMMENTED** [review]
Thank you for this! I have one suggested change, otherwise LGTM.

File: bin/functions.sh (hunk 1 above)
Comment (jeanfabrice) [reply bin/functions.sh]: pushed 5aaeddf to improve this logic

**Comment by JohnMcLear** [comment]
Ready for review?  It's still flagged as WIP:

**Comment by jeanfabrice** [comment]
> Ready for review? It's still flagged as WIP:

'cause I'm not able to address the Windows part :/

**Pull request merged**